src/main/java/swinglib/ErrorMessageBuilder.java

```diff
@@ -0,0 +1,117 @@
+package swinglib;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import java.awt.GraphicsEnvironment;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.swing.JOptionPane;
+
+import org.apache.commons.math3.util.Pair;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+
+/**
+ * Swing builder for an error message pop-up (Uses JOptionPane). Example usages:
+ * <code><pre>
+ *   JOptionPaneBuilder.builder()
+ *     .title("Something Crashed")
+ *     .message("Error Details")
+ *     .option("ok", () -> {})
+ *     .option("quit", () -> System.exit(0))
+ *     .buildAndShow();
+ * </pre></code>
+ * Default error dialog with an 'ok' button:
+ * <code><pre>
+ *   JOptionPaneBuilder.builder()
+ *     .message("error details")
+ *     .buildAndShow();
+ * </pre></code>
+ */
+public class ErrorMessageBuilder {
+
+  private String title = "Error";
+  private String message;
+  private List<Pair<String, Runnable>> actions = new ArrayList<>();
```

Comment:
Ordering is important, so can't use a `Map` a `Collection`.

Reply:
A LinkedHashMap preserves the insertion order.
The only "difficultty" would be to reverse the order when converting it to an arr.

Reply:
Some client example code to consider:
```
.option("ok", () -> {})
.option("ok", () -> {})
```
The question is whether that should produce 2 buttons or 1, and it's also a question if de-duplication is possible. For first part, I think it's more straight forward for builder code like that to add two buttons insead of de-duping. Second part, can we de-dupe even? We need an equality for the runnable, that is very difficult.

To that extent, I do think initially that `ArrayList` is most straight forward. `Map` implies we are checking for dupes, but it's never going to be the case that we would have duplicate objects. It might be a bit deceiving I think to imply that we can even detect duplicates.

Reply:
I'm convinced now ^^

Reply:
Updating the `Pair` class to be a `CloseButton` should actually make this point moot. We have `List<CloseButton>` instead of the suggested `Map<String, Runnable>`. I do want a `List` API on the left hand side, not a `Map`, so converting to a `Map<> = LinkedHashMap` I do not think is best. Ordering is important as it has implications for what a client can expect when inserting multiple option buttons. Any attempt at de-duplication or sorting I think would be side effects, not obvious from the caller and would put into question how the API works.

If we did want to allow an ordering to be specified other than insertion ordering left to right, we probably should do something like:
```
.option("ok", () -> {})
.option("ok", () -> {})
.orderOptionsBy(SortOptions.INSERTION_LEFT_TO_RIGHT) // default value
.orderOptionsBy(SortOptions.ALPHA)
.orderOptionsBy(SortOptions.INSERTION_RIGHT_TIO_LEFT)
```
I don't think there is  use-case for alpha or right to left insertion, so YAGNI on that until we need it.

Reply:
Summarizing - 9c0e9e9 should address this. Interesting data representation consideration :) 
 ; though I'd like to stick with a `List`